python/src/dataset.rs

```diff
@@ -1367,15 +1367,34 @@ impl Dataset {
     }
 
     #[staticmethod]
-    #[pyo3(signature = (dest, storage_options = None))]
-    fn drop(dest: String, storage_options: Option<HashMap<String, String>>) -> PyResult<()> {
+    #[pyo3(signature = (dest, storage_options = None, ignore_not_found = true))]
+    fn drop(
+        dest: String,
+        storage_options: Option<HashMap<String, String>>,
+        ignore_not_found: bool,
+    ) -> PyResult<()> {
         RT.spawn(None, async move {
             let (object_store, path) =
                 object_store_from_uri_or_path(&dest, storage_options).await?;
-            object_store
-                .remove_dir_all(path)
-                .await
-                .map_err(|e| PyIOError::new_err(e.to_string()))
+            let result = object_store.remove_dir_all(path).await;
+
+            match result {
+                Ok(_) => Ok(()),
+                Err(e) => {
+                    let is_not_found = matches!(&e, lance_core::Error::NotFound { .. });
+
+                    if ignore_not_found && is_not_found {
+                        Ok(())
+                    } else if is_not_found {
+                        Err(PyFileNotFoundError::new_err(format!(
```

Comment:
This change is technically backwards compatible since we will return `PyFileNotFoundError` instead of `PyIOError`.  I'm not sure if it matters too much.

Reply:
Ah, never mind, `FileNotFoundError` extends `OSError` (which is an alias with `IOError` 🫠)